pkg/util/fs_util.go

```diff
@@ -507,9 +507,18 @@ func CopyDir(src, dest, buildcontext string, chownUID, chownGid int) ([]string,
 		if fi.IsDir() {
 			logrus.Debugf("Creating directory %s", destPath)
 
-			if err := os.MkdirAll(destPath, fi.Mode()); err != nil {
+			// Make any parent directories with 0755 permissions
+			if err := os.MkdirAll(dest, 0755); err != nil {
```

Comment:
Don't we need to if `fi` is directory or file?

Reply:
`dest` is the parent directory of where the file is going to be copied, and also this is in the fi.IsDir path already so destPath is also a directory.  the CopyFile/CopySymlink methods correctly create parent directories with 0755 permissions if needed

Reply:
works for me